src/Symfony/Component/Security/Core/Security.php

```diff
@@ -30,8 +32,14 @@ class Security implements AuthorizationCheckerInterface
 
     private $container;
 
-    public function __construct(ContainerInterface $container)
+    /**
+     * @var array
+     */
+    private $authenticators;
+
+    public function __construct(array $authenticators, ContainerInterface $container)
```

Comment:
Should `$authenticators` be an optional second parameter rather? That way changes to this core class can be backwards compatible, which would avoid necessitating a major release.

Reply:
Thanks ! Changes done.